src/sentry/monitors/consumers/check_in.py

```diff
@@ -77,7 +78,11 @@ def _ensure_monitor_with_config(
 
     # Update existing monitor
     if monitor and not created and monitor.config != validated_config:
-        monitor.update(config=validated_config)
+        monitor_config = deepcopy(monitor.config)
```

Comment:
Might be worth making this a function on the monitor model instead of duplicating it around.